### PR TITLE
implements HolidayName().

### PR DESCRIPTION
**Pull request by shirou**
I found `holiday` is not implemented yet. This PR implements `HolidayName` that returns Holiday Name as exported function.

### CODE REVIEW

File: holidayjp.go [hunk 1]
```diff
@@ -10,17 +11,22 @@ import (
...
 }
 
-// Holiday returns Holiday name of the given time.
-// If it's not holiday, returns error.
-func holiday(t time.Time) (string, error) {
-	// TODO
-	return "", nil
+// HolidayName returns Holiday name of the given time.
+// If it's not holiday, returns error(ErrNotHoliday).
+func HolidayName(target time.Time) (string, error) {
```
Comment (tcnksm) [inline holidayjp.go]: I prefer `Holiday()` for this function instead of `HolidayName()`

File: holidayjp.go [hunk 2]
```diff
@@ -10,17 +11,22 @@ import (
 
 //go:generate go-bindata -pkg=asset -o asset/asset.go ./asset/holidays.yml
 
+var ErrNotHoliday = fmt.Errorf("not holiday")
```
Comment (tcnksm) [inline holidayjp.go]: I want to avoid to define own error type.

Comment (tcnksm) [inline holidayjp.go]: Can you just return error?, 

``` bash
return "", fmt.Errorf("holiday is not found")
```

**Comment by tcnksm** [comment]
Thanks, I put some comments

**Comment by shirou** [comment]
Thank you for these comments!

I updated as these,
1. rename function name to `Holiday`
2. use `fmt.Errorf`

**Comment by tcnksm** [comment]
Great! thanks

**Pull request merged**